content/guides/end-to-end-testing/okta-authentication.md

```diff
@@ -70,11 +76,116 @@ require('dotenv').config()
+4. cache the results with [`cy.session()`](/api/commands/session)
+
+```jsx
+// cypress/support/auth-provider-commands/okta.ts
+// Okta
+const loginToOkta = (username: string, password: string) => {
+  Cypress.log({
+    displayName: 'OKTA LOGIN',
+    message: [`🔐 Authenticating | ${username}`],
+    autoEnd: false,
+  })
+
+  cy.visit('/')
+  cy.origin(
+    Cypress.env('okta_domain'),
+    { args: { username, password } },
+    ({ username, password }) => {
+      cy.get('input[name="identifier"]').type(username)
+      cy.get('input[name="credentials.passcode"]').type(password, {
+        log: false,
+      })
+      cy.get('[type="submit"]').click()
+    }
+  )
+
+  cy.get('[data-test="sidenav-username"]').should('contain', username)
+}
+// right now our custom command is light. More on this later!
+Cypress.Commands.add('loginByOkta', (username: string, password: string) => {
+  return loginToOkta(username, password)
+})
+```
+
+Secondly, we can use our `loginByOkta` command in the test. Below is our test to
+login as a user via [Okta](https://okta.com) and run a few basic sanity tests.
+
+<Alert type="success">
+
+The
+[runnable version of this test](https://github.com/cypress-io/cypress-realworld-app/blob/develop/cypress/tests/ui-auth-providers/okta.spec.ts)
+is in the
+[Cypress Real World App](https://github.com/cypress-io/cypress-realworld-app).
+
+</Alert>
+
+```jsx
+describe('Okta', function () {
+  beforeEach(function () {
+    cy.task('db:seed')
+    cy.loginByOkta(Cypress.env('okta_username'), Cypress.env('okta_password'))
+  })
+
+  it('verifies signed in user does not have a bank account', function () {
+    cy.get('[data-test="sidenav-bankaccounts"]').click()
+    cy.get('[data-test="empty-list-header"]').should('be.visible')
+  })
+})
+```
+
+<DocsVideo src="/img/examples/okta-origin.mp4"></DocsVideo>
+
+Lastly, we can refactor our login command to take advantage of
+[`cy.session()`](/api/commands/session) to store our logged in user so we don't
+have to reauthenticate with everything test.
+
+```jsx
+Cypress.Commands.add('loginByOkta', (username: string, password: string) => {
+  cy.session(
+    `okta-${username}`,
+    () => {
+      return loginToOkta(username, password)
+    },
+    {
+      validate() {
+        cy.visit('/')
+        cy.get('[data-test="sidenav-username"]').should('contain', username)
+      },
+    }
+  )
+})
+```
+
+<DocsVideo src="/img/examples/okta-session-restore.mp4"></DocsVideo>
+
+### Programmatic Login
+
 Next, we will write a command named `loginByOktaApi` to perform a programmatic
 login into [Okta](https://okta.com) and set an item in localStorage with the
 authenticated users details, which we will use in our application code to verify
 we are authenticated under test.
 
+In order to make sure this is enabled inside the
+[Cypress Real World App](https://github.com/cypress-io/cypress-realworld-app),
+Please set the `REACT_APP_OKTA_PROGRAMMATIC` environment variable to `true`.
```

Comment:
Why is this ENV needed? Also do we need to call it out here or can we link to a section in the read me to share how to set this up?

Reply:
I added `REACT_APP_OKTA_PROGRAMMATIC` since the programmatic approach strips the auth guards out of the webapp, which we don't want with UI login with `cy.origin()` since we want to test how the app actually behaves and get the natural redirect to the auth page when we aren't authenticated

Reply:
I figured here is fine since the README just points back to the guide